src/main/java/com/bean00/server/RequestProcessor.java

```diff
@@ -17,27 +17,64 @@ public RequestProcessor(DataStore dataStore) {
     }
 
     public Response processRequest(Request request) throws IOException {
+        String requestMethod = request.getRequestMethod();
+        Response response;
+
+        if (requestMethod.equals(Method.PUT)) {
+            response = putHandler(request);
+        } else if (requestMethod.equals(Method.GET) || requestMethod.equals(Method.HEAD)) {
+            response = getHeadHandler(request);
+        } else {
+            response = methodNotAllowedResponse("GET, HEAD, PUT");
+        }
+
+        return response;
+    }
+
+    private Response putHandler(Request request) throws IOException {
+        String requestURL = request.getRequestURL();
+        byte[] requestBody = request.getBodyAsBytes();
+
+        if (dataStore.isDirectory(requestURL)) {
+            return methodNotAllowedResponse("GET, HEAD");
```

Comment:
Why not use Method.GET and Method.HEAD here?

Reply:
Addressed. I've added a helper method to build up the `allowedMethodsString`.